src/cli_config.rs

```diff
@@ -0,0 +1,83 @@
+use std::net::ToSocketAddrs;
+
+use clap::{Arg, App, AppSettings};
+use hex::FromHex;
+use itertools::Itertools;
+use num_cpus;
+use tox::toxcore::crypto_core::*;
+use tox::toxcore::dht::packed_node::PackedNode;
+
+/// Config parsed from command line arguments.
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct CliConfig {
+    /// List of bootstrap nodes.
+    pub bootstrap_nodes: Vec<PackedNode>,
+    /// Number of threads for execution. None if single threaded runtime should
+    /// be used.
+    pub threads_count: Option<usize>,
+}
+
+/// Parse command line arguments.
+pub fn cli_parse() -> CliConfig {
+    let num_cpus_string = num_cpus::get().to_string();
+
+    let matches = App::new(crate_name!())
+        .version(crate_version!())
+        .author(crate_authors!("\n"))
+        .about(crate_description!())
+        .setting(AppSettings::ColoredHelp)
+        .arg(Arg::with_name("bootstrap-node")
+            .short("b")
+            .long("bootstrap-node")
+            .help("Node to perform initial bootstrap")
+            .multiple(true)
+            .takes_value(true)
+            .number_of_values(2)
+            .value_names(&["public key", "address"]))
+        .arg(Arg::with_name("threaded")
```

Comment:
See issue title. The arg should be: `-j, --threads`,

> -1 (default) = singlethreaded runtime, 0 = multithreaded with threads=num cpu, 1,2,3...N = multithreaded with N threads

Reply:
It doesn't say anything about magic `-1`, `0`. And as I said `-j` means `--jobs` there, it doesn't have `--threads`.

Reply:
I do really think that `-j` is quite often used as the number of threads. See https://github.com/BurntSushi/ripgrep/blob/master/src/app.rs#L1546 for example.

Reply:
ok, I can change it to `-j` then. But do you agree about separate flag for multithreading?